modules/masterbar/masterbar.php

```diff
@@ -68,6 +76,104 @@ function __construct() {
+			'wp-admin-bar-plugins'                    => array( 'masterbar_configure_plugins_link_click', 'https://wordpress.com/plugins/' . $site_slug ),
+			'wp-admin-bar-plugins-secondary'          => array( 'masterbar_configure_plugins_add_button_click', 'https://wordpress.com/plugins/browse/' . $site_slug ),
+			'wp-admin-bar-blog-settings'              => array( 'masterbar_settings_link_click', 'https://wordpress.com/settings/general/' . $site_slug ),
+
+			//reader
+			'wp-admin-bar-following'                  => array( 'masterbar_followed_sites_link_click', 'https://wordpress.com' ),
+			'wp-admin-bar-following-secondary'        => array( 'masterbar_followed_sites_manage_button_click', 'https://wordpress.com/following/edit' ),
+			'wp-admin-bar-discover-discover'          => array( 'masterbar_reader_discover_link_click', 'https://wordpress.com/discover' ),
+			'wp-admin-bar-discover-search'            => array( 'masterbar_reader_search_link_click', 'https://wordpress.com/read/search' ),
+			'wp-admin-bar-discover-recommended-blogs' => array( 'masterbar_reader_recommendations_link_click', 'https://wordpress.com/recommendations' ),
+			'wp-admin-bar-my-activity-my-likes'       => array( 'masterbar_reader_my_links_link_click', 'https://wordpress.com/activities/likes' ),
+
+			//account
+			'wp-admin-bar-user-info'                  => array( 'masterbar_user_name_link_click', 'https://wordpress.com' ),
+			// account - profile
+			'wp-admin-bar-my-profile'                 => array( 'masterbar_profile_my_profile_link_click', 'https://wordpress.com/me' ),
+			'wp-admin-bar-account-settings'           => array( 'masterbar_profile_account_settings_link_click', 'https://wordpress.com/me/account' ),
+			'wp-admin-bar-billing'                    => array( 'masterbar_profile_manage_purchases_link_click', 'https://wordpress.com/me/purchases' ),
+			'wp-admin-bar-security'                   => array( 'masterbar_profile_security_link_click', 'https://wordpress.com/me/security' ),
+			'wp-admin-bar-notifications'              => array( 'masterbar_profile_notifications_link_click', 'https://wordpress.com/me/notifications' ),
+			//account - special
+			'wp-admin-bar-get-apps'                   => array( 'masterbar_get_apps_link_click', 'https://wordpress.com/me/get-apps' ),
+			'wp-admin-bar-next-steps'                 => array( 'masterbar_next_steps_link_click', 'https://wordpress.com/me/next' ),
+			'wp-admin-bar-help'                       => array( 'masterbar_help_link_click', 'https://jetpack.com/support/' ),
+		);
+	}
+
+	function tracks_and_redirect_allow_wpcom_domain( $domains ) {
+		if ( empty( $domains ) ) {
+			$domains = array();
+		}
+		$domains[] = 'wordpress.com';
+		return array_unique( $domains );
+	}
+
+	public function tracks_and_redirect_add_route() {
+		add_rewrite_rule( 'jetpack-track-and-bounce.php\?tracks_and_bounce=(\w*)', 'index.php?tracks_and_bounce=$matches[1]', 'top' );
```

Comment:
Question - what is the rewrite rule helping us to accomplish here? Is it simply to allow us to have a more descriptive and explicit name for our urls?

Reply:
yeah, basically the idea is to have an explicit address, but It's doing more harm than good. I'll replace it...

Reply:
If you think it needs to be replaced, go for it 👍  
I was just trying to understand what it was for 😄